code/game/supplyshuttle.dm

```diff
@@ -46,6 +46,192 @@ var/datum/controller/supply/supply_controller = new()
+	name = "supply shuttle"
+	id = SHUTTLE_SUPPLY
+	callTime = 15 SECONDS
+
+	dir = WEST
+	port_direction = EAST
+	width = 5
+	dwidth = 2
+	dheight = 2
+	height = 5
+	movement_force = list("KNOCKDOWN" = 0, "THROW" = 0)
+	use_ripples = FALSE
+	var/list/gears = list()
+	var/list/obj/structure/machinery/door/poddoor/railing/railings = list()
+	///The faction of this docking port (aka, on which ship it is located)
+	var/faction = FACTION_MARINE
+	/// Id of the home docking port
+	var/home_id = "supply_home"
+
+/obj/docking_port/mobile/supply/Destroy(force)
+	for(var/i in railings)
+		var/obj/structure/machinery/door/poddoor/railing/railing = i
+		railing.linked_pad = null
+	railings.Cut()
+	return ..()
+
+
+/obj/docking_port/mobile/supply/afterShuttleMove()
+	. = ..()
+	if(getDockedId() == home_id)
+		for(var/j in railings)
+			var/obj/structure/machinery/door/poddoor/railing/R = j
+			R.open()
+
+/obj/docking_port/mobile/supply/on_ignition()
+	if(getDockedId() == home_id)
+		for(var/j in railings)
+			var/obj/structure/machinery/door/poddoor/railing/R = j
+			R.close()
+		for(var/i in gears)
+			var/obj/structure/machinery/gear/G = i
+			G.start_moving(NORTH)
+	else
+		for(var/i in gears)
+			var/obj/structure/machinery/gear/G = i
+			G.start_moving(SOUTH)
+
+/obj/docking_port/mobile/supply/register()
+	. = ..()
+	for(var/obj/structure/machinery/gear/G in GLOB.machines)
+		if(G.id == "supply_elevator_gear")
+			gears += G
+			RegisterSignal(G, COMSIG_PARENT_QDELETING, .proc/clean_gear)
+	for(var/obj/structure/machinery/door/poddoor/railing/R in GLOB.machines)
+		if(R.id == "supply_elevator_railing")
+			railings += R
+			RegisterSignal(R, COMSIG_PARENT_QDELETING, .proc/clean_railing)
+			R.linked_pad = src
+			R.open()
+
+///Signal handler when a gear is destroyed
+/obj/docking_port/mobile/supply/proc/clean_gear(datum/source)
+	SIGNAL_HANDLER
+	gears -= source
+
+///Signal handler when a railing is destroyed
+/obj/docking_port/mobile/supply/proc/clean_railing(datum/source)
+	SIGNAL_HANDLER
+	railings -= source
+
+/obj/docking_port/mobile/supply/canMove()
+	if(is_station_level(z))
+		return check_blacklist(shuttle_areas)
+	return ..()
+
+/obj/docking_port/mobile/supply/proc/check_blacklist(areaInstances)
+	if(!areaInstances)
+		areaInstances = shuttle_areas
+	for(var/place in areaInstances)
+		var/area/shuttle/shuttle_area = place
+		for(var/trf in shuttle_area)
+			var/turf/T = trf
+			for(var/a in T.GetAllContents())
+				if(isxeno(a))
+					var/mob/living/L = a
+					if(L.stat == DEAD)
+						continue
+				if(ishuman(a))
+					var/mob/living/carbon/human/human_to_sell = a
+					if(human_to_sell.stat == DEAD && can_sell_human_body(human_to_sell, faction))
+						continue
+				if(is_type_in_typecache(a, GLOB.blacklisted_cargo_types))
+					return FALSE
+	return TRUE
+
+/obj/docking_port/mobile/supply/request(obj/docking_port/stationary/S)
+	if(mode != SHUTTLE_IDLE)
+		return 2
+	return ..()
+
+/obj/docking_port/mobile/supply/proc/buy(mob/user)
```

Comment:
Again, I would recommend splitting out the logic of what the supply shuttle does into somewhere else, either a datum or even its own controller if need be. The shuttle doesn't need to be so tightly coupled to the supply system